include/transport/priority_queue.h

```diff
@@ -160,8 +159,26 @@ namespace qtransport {
         }
 
         // TODO: Consider changing empty/size to look at timeQueue sizes - maybe support blocking pops
-        size_t size() const { return _queue.size(); }
-        bool empty() const { return _queue.empty(); }
+        size_t size() const {
+            size_t sz=0;
+            for (size_t i = 0; i < _queue.size(); i++) {
+                if (_queue[i]) {
+                    sz += _queue[i]->size();
+                }
+            }
+
+            return sz;
+        }
+
+        bool empty() const {
+            for (size_t i = 0; i < _queue.size(); i++) {
+                if (_queue[i] && !_queue[i]->empty()) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
```

Comment:
Alternatively, we could let STL do it for us:
```suggestion
        bool empty() const {
            return std::all_of(_queue.begin(), _queue.end(), [&](const auto& elem) { return elem->empty(); });
        }
```

Reply:
Specifically from [std::all_of](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of):
> 1-2) true if unary predicate returns true for all elements in the range, false otherwise. Returns true if the range is empty.